<launch>
   <node name="ekf_odom" pkg="robot_localization" type="ekf_localization_node">  
    <param name="frequency" value="10." />
    <param name="sensor_timeout" value="0.5" />
    <param name="two_d_mode" value="false" />
    <param name="frequency" value="10" />
    <param name="transform_time_offset" value="-0.02" />

    <param name="map_frame" value="map" />
    <param name="odom_frame" value="odom" />
    <param name="base_link_frame" value="base_link" />
    <param name="world_frame" value="odom" />

    <param name="twist0" value="/wheel_odom_cov" />
    <param name="twist0_differential" value="false"/>
    <param name="twist0_relative" value="true"/>
    <param name="twist0_queue_size" value="5"/>
    <rosparam param="twist0_config">
      [false, false, false, 
      false, false, false,
      true, true, false, 
      false, false, true,
      false, false, false]</rosparam>

    <param name="imu0" value="/zed2/imu/data" />
    <param name="imu0_differential" value="false"/>
    <param name="imu0_relative" value="true"/>
    <param name="imu0_queue_size" value="5"/>
    <param name="imu0_remove_gravitational_acceleration" value="true"/>
    <rosparam param="imu0_config">
      [false, false, false, 
      true, true, true,
      false, false, false, 
      true, true, true,
      false, false, false]</rosparam>

  </node>

  <node name="ekf_map" pkg="robot_localization" type="ekf_localization_node">  
    <param name="frequency" value="10." />
    <param name="sensor_timeout" value="0.5" />
    <param name="two_d_mode" value="false" />
    <param name="frequency" value="10" />

    <param name="map_frame" value="map" />
    <param name="odom_frame" value="odom" />
    <param name="base_link_frame" value="base_link" />
    <param name="world_frame" value="map" />

    <param name="twist0" value="/wheel_odom_cov" />
    <param name="twist0_differential" value="false"/>
    <param name="twist0_relative" value="false"/>
    <param name="twist0_queue_size" value="5"/>
    <rosparam param="twist0_config">
      [false, false, false, 
      false, false, false,
      true, true, false, 
      false, false, true,
      false, false, false]</rosparam>

    <param name="imu0" value="/zed2/imu/data" />
    <param name="imu0_differential" value="false"/>
    <param name="imu0_relative" value="true"/>
    <param name="imu0_queue_size" value="1"/>
    <param name="imu0_remove_gravitational_acceleration" value="true"/>
    <rosparam param="imu0_config">
      [false, false, false, 
      true, true, true,
      false, false, false, 
      true, true, true,
      false, false, false]</rosparam>

  </node>
  
  <node name="publish_pose" pkg="topic_tools" type="transform" args="/odometry/filtered /odometry/filtered/pose geometry_msgs/PoseWithCovarianceStamped 'm.header, m.pose'" />

</launch>